lib/internal/util/comparisons.js

```diff
@@ -230,10 +230,14 @@ function innerDeepEqual(val1, val2, mode, memos) {
   } else if (val1Tag === '[object Object]') {
     return keyCheck(val1, val2, mode, memos, kNoIterator);
   } else if (isDate(val1)) {
-    if (!isDate(val2) ||
-        DatePrototypeGetTime(val1) !== DatePrototypeGetTime(val2)) {
+    if (!isDate(val2)) {
       return false;
     }
+    const time1 = DatePrototypeGetTime(val1);
+    const time2 = DatePrototypeGetTime(val2);
+    if (time1 !== time2) {
+      return NumberIsNaN(time1) && NumberIsNaN(time2);
```

Comment:
might be faster to do:
```suggestion
      return time1 !== time1 && time2 !== time2;
```

Reply:
The performance is AFAIK identical and it's more difficult to read (most people don't understand it and see it as a mistake). I can briefly check that again.

Reply:
cool, if that's the case then pls keep the NumberIsNaN and resolve :-)

Reply:
I was wrong. This is indeed still about 7-8% faster.

I guess it's something that the @nodejs/v8 team might want to look into.

Reply:
`NumberIsNaN` is a function call, `!==` is an inline check. V8 _can_ inline `Number.isNaN` in the right circumstances but not (confidence level > 50%) when it's copied to another object.